src/server/shell/basic_accessibility_manager.cpp

```diff
@@ -103,11 +103,10 @@ void mir::shell::BasicAccessibilityManager::mousekeys_enabled(bool on)
         main_loop->spawn([this]
         {
             auto state = this->mutable_state.lock();
-            if (!state->mousekey_pointer)
+            if (!transformer.is_registered())
```

Comment:
The point of `InputEventTransformer::Registration` was to avoid the need for BAM to track registrations explicitly. Adding a wrapper around that so that we can test `is_registered()` rather defeats the point

Reply:
> The wrapper isn't for `is_registered`, I could also do `std::optional::has_value` to the same effect. The problem `is_registered` solves here is covering for shell author errors such as disabling mousekeys when already disabled or double-enabling it. Later on once #3873 is implemented, we'll also need to track registrations to enable or disable the virtual device.

I agree that `std::optional::has_value` is equivalent. But the idea of `InputEventTransformer::Registration`
to avoid tracking registrations. _Something_ is wrong if are also adding code to track registrations.